src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double GROW_MULTIPLYER = 1.5;
+    private static final int DEFAUL_SIZE = 10;
+    private int size = 0;
+    private T[] elements;
+
+    public ArrayList() {
+        this.elements = (T[]) new Object[DEFAUL_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        resize();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of array boundaries");
+        }
+        resize();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            resize();
+            elements[size] = list.get(i);
+            size++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        final T res = elements[index];
+        System.arraycopy(elements, index + 1, elements, index, size - index - 1);
+        elements[size - 1] = null;
+        size--;
```

Comment:
```suggestion
        elements[--size] = null;
```
use pre increment in situations like this

Reply:
not fixed